src/remote/sigmsg.rs

```diff
@@ -0,0 +1,757 @@
+//! Signed Message CMS wrappers used in the RPKI publication (RFC 8181) and
+//! provisioning (RFC 6492) protocols.
+
+use bcder::encode::PrimitiveContent;
+use bcder::{decode, encode, Captured};
+use bcder::{Mode, Oid, OctetString, Tag, xerr};
+use bytes::Bytes;
+use crate::repository::crl::RevokedCertificates;
+use crate::repository::crypto::{
+    DigestAlgorithm, KeyIdentifier, Signature, SignatureAlgorithm, Signer,
+    SigningError
+};
+use crate::repository::oid::{self, PROTOCOL_CONTENT_TYPE};
+use crate::repository::sigobj::{
+    MessageDigest, SignedAttrs
+};
+use crate::repository::x509::{
+    Name, Serial, SignedData, Time, ValidationError,
+    update_once, Validity, encode_extension
+};
+use super::idcert::IdCert;
+
+/// A signed message, as used in RPKI remote protocols.
+/// 
+/// This is a *lot* like [`SignedObject`], i.e. Signed Object Template
+/// for the RPKI (RFC 6488), but with subtle differences which make using
+/// a common code base for both types somewhat painful. Hence, this separate
+/// but similar structure.
+/// 
+/// Most important differences to watch out for:
+/// = This uses [`IdCert`] rather then (resource) [`Cert`] as the EE
+/// = This includes a CRL
+/// 
+/// The latter seems rather pointless in case one uses single use EE
+/// certificates, but it's required by the protocol.. So, when sending
+/// we just include an empty CRL to go with our short-lived single-use
+/// EE certificates that will never be revoked.
+#[derive(Clone, Debug)]
+pub struct  SignedMessage {
+    //--- From SignedData
+    //
+    digest_algorithm: DigestAlgorithm,
+    content_type: Oid<Bytes>,
+    content: OctetString,
+    
+    // Theoretically there could be multiple certificates, i.e. a chain
+    // of certificates needed for validation to the known trusted TA
+    // certificate for the other party, but in practice no one does this.
+    //
+    // So in our case we insist on a single embedded EE certificate which
+    // is expected to be signed directly under the known TA certificate.
+    ee_cert: IdCert,
+
+    // Similarly there could be one CRL for each embedded certificate, but
+    // since we just support a single EE certificate here we also expect a
+    // single Crl.
+    crl: SignedMessageCrl,
+
+    //--- From SignerInfo
+    //
+    sid: KeyIdentifier,
+    signed_attrs: SignedAttrs,
+    signature: Signature,
+
+    //--- SignedAttributes
+    //
+    message_digest: MessageDigest,
+    _signing_time: Option<Time>,
```

Comment:
Add accessor methods to avoid the dead code? Or drop them altogether?

Reply:
removed the fields for now - they are still in the signed attributes if we would need them.